### PR TITLE
Fixes to subnav

### LINKED ISSUE #3025: Content bug: empty Subnav sections on many pages
## What page(s) did you find the problem on?

A non-exhaustive list:

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles#Subnav
https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Grid_Layout/Subgrid#subnav (the one I first noticed)
https://developer.mozilla.org/en-US/docs/Web/XPath#subnav
https://developer.mozilla.org/en-US/docs/MDN/At_ten#Subnav

## What is the problem?

These sections are often empty, but not always. It seems like these are the remnants of some previous system where they weren't visible, probably being extracting into a sidebar or menu.

## What did you expect to see?

The empty sections should be removed. But it's not clear how to do that, because they're often followed by other content, the Quick links, and sometimes the Subnav is in a `<div class="hidden">`. What's the right pattern?

### PR DESCRIPTION
**Pull request by hamishwillee**
Fixes #3025

This 
- removes the subnav **heading**, which does nothing. 
- removes quicklinks sections where there is a macro that is already being displayed instead. 
- For CSS topics there was a mix of quicklinks and sidebar macro usage. I have for this case further made all quicklinks to use macro instead.

### CODE REVIEW

File: files/en-us/mdn/at_ten/index.html [hunk 1]
```diff
@@ -23,14 +21,11 @@ <h2 id="Contributing_to_MDN">Contributing to MDN</h2>
...
 
 
-</div>
-
-<div class="column-4">{{TenthCampaignQuote}}</div>
-
-<h2 id="Subnav">Subnav</h2>
+{{TenthCampaignQuote}}
 
+<section id="Quick_links">
```
Comment (hamishwillee) [inline files/en-us/mdn/at_ten/index.html:26]: FYI removed layout crap in old doc. Also turned the "subnav" into quicklinks, removing the link back to this doc.

File: files/en-us/web/accessibility/aria/roles/index.html [hunk 2]
```diff
@@ -77,7 +77,7 @@
 </div>
 
 
-<h6 id="Subnav">Subnav</h6>
```
Comment (hamishwillee) [inline files/en-us/web/accessibility/aria/roles/index.html:80]: Removed heading, not used.

File: files/en-us/web/api/document_object_model/events/index.html [hunk 3]
```diff
@@ -75,11 +80,3 @@ <h2 id="Accessing_Event_interfaces">Accessing Event interfaces</h2>
 // any function should have an appropriate name, that's what called semantic
 table_el.onclick = print
 </pre>
-
```
Comment (hamishwillee) [inline files/en-us/web/api/document_object_model/events/index.html:78]: FYI only relevant change in this is to remove this subnav section.  The page already has a sidebar macro so this was just ignored.

File: files/en-us/web/api/document_object_model/examples/index.html [hunk 4]
```diff
@@ -5,6 +5,7 @@
   - DOM
   - DOM Reference
 ---
+<div>{{DefaultAPISidebar("DOM")}}</div>
```
Comment (hamishwillee) [inline files/en-us/web/api/document_object_model/examples/index.html:8]: Added standard DOM sidebar and removed the subnav section below.

File: files/en-us/web/api/document_object_model/introduction/index.html [hunk 5]
```diff
@@ -10,6 +10,7 @@
   - Introduction
   - Tutorial
 ---
+<div>{{DefaultAPISidebar("DOM")}}</div>
```
Comment (hamishwillee) [inline files/en-us/web/api/document_object_model/introduction/index.html:13]: FYI Added standard DOM sidebar and removed the subnav section below.

File: files/en-us/web/css/css_grid_layout/subgrid/index.html [hunk 6]
```diff
@@ -123,73 +124,3 @@ <h2 id="See_also">See also</h2>
  <li><a href="https://noti.st/rachelandrew/i6gUcF/hello-subgrid">Hello Subgrid!</a> A presentation from CSSConf.eu</li>
 </ul>
 
-<h2 id="Subnav">Subnav</h2>
```
Comment (hamishwillee) [inline files/en-us/web/css/css_grid_layout/subgrid/index.html:126]: FYI removed subnav - not rendered as the page has a sidebar macro (up top). FYI this duplicated sidebar macro output.

File: files/en-us/web/exslt/index.html [hunk 7]
```diff
@@ -5,7 +5,6 @@
   - EXSLT
   - Landing
 ---
-<p>{{XSLTRef}}</p>
```
Comment (hamishwillee) [inline files/en-us/web/exslt/index.html:8]: This one a little odd. I thought at first this was a sidebar macro but it just imported this line. I don't see it as useful here

image

Below removed the subnav heading, but kept the quick links so we get sidebar.

File: files/en-us/web/css/justify-tracks/index.html [hunk 8]
```diff
@@ -10,6 +10,8 @@
   - justify-tracks
   - masonry
 ---
+<div>{{CSSRef}}</div>
```
Comment (hamishwillee) [inline files/en-us/web/css/justify-tracks/index.html:13]: Replaced custom subnav with macro sidebar

File: files/en-us/web/xpath/index.html [hunk 9]
```diff
@@ -62,10 +58,6 @@ <h2 id="Related_Topics">Related Topics</h2>
  <li><a href="/en-US/docs/Web/XSLT">XSLT</a>, <a href="/en-US/docs/Web/XML">XML</a>, <a href="/en-US/docs/Web/API/Document_Object_Model">DOM</a></li>
  <li><a href="/en-US/docs/Web/XPath/Comparison_with_CSS_selectors">Comparison of CSS Selectors and XPath</a></li>
 </ul>
-</div>
-</div>
-
-<h2 id="Subnav">Subnav</h2>
```
Comment (hamishwillee) [inline files/en-us/web/xpath/index.html:68]: FYI just removed heading. The sidebar still present.

**Review by chrisdavidmills — APPROVED** [review]
Some great work here; thanks @hamishwillee !

**Comment by peterbe** [comment]
Well done!

**Pull request merged**